sdk/database/helper/connutil/sql.go

```diff
@@ -107,6 +135,21 @@ func (c *SQLConnectionProducer) Init(ctx context.Context, conf map[string]interf
 		return nil, errwrap.Wrapf("invalid max_connection_lifetime: {{err}}", err)
 	}
 
+	if c.AuthType == AuthTypeGCPIAM {
```

Comment:
I had mentioned this in a comment above, but I think that we can get rid of the global driver map and the lock if we make use of `c.Initialized` as a singleton check.

```suggestion
	if c.AuthType == AuthTypeGCPIAM && !c.Initialized {
```

Reply:
My understanding was that each connection producer instance was tied to a config, so we shouldn't need any init check?

Reply:
Resolving this conversation since global driver map was removed